src/interpreter/inner.rs

```diff
@@ -216,47 +228,36 @@ pub(super) fn from_txdata<'txin>(
                     None, // Tr script code None
                 ))
             } else {
-                // wit_stack.len() >=2
-                // Check for annex
                 let ctrl_blk = wit_stack.pop().ok_or(Error::UnexpectedStackEnd)?;
                 let ctrl_blk = ctrl_blk.as_push()?;
                 let tap_script = wit_stack.pop().ok_or(Error::UnexpectedStackEnd)?;
-                if ctrl_blk.len() > 0 && ctrl_blk[0] == TAPROOT_ANNEX_PREFIX {
-                    // Annex is non-standard, bitcoin consensus rules ignore it.
-                    // Our sighash structure and signature verification
-                    // does not support annex, return error
-                    return Err(Error::TapAnnexUnsupported);
```

Comment:
you should keep this, no?

Reply:
Yes, indeed. Initially, I was thinking that we can just pop and ignore it. But as the comment says here, since we don't support signature verification with Annex, it is correct to Error